buildSrc/src/main/groovy/org/elasticsearch/gradle/DependenciesInfoTask.groovy

```diff
@@ -0,0 +1,177 @@
+
+    /** Dependencies to gather information from. */
+    @Input
+    public DependencySet dependencies
+
+    /** Directory to read license files */
+    @InputDirectory
+    public File licensesDir = new File(project.projectDir, 'licenses')
+
+    @OutputFile
+    File outputFile = new File(project.buildDir, "reports/dependencies/dependencies.csv")
+
+    public DependenciesInfoTask() {
+        description = 'Create a CSV file with dependencies information.'
+    }
+
+    @TaskAction
+    public void generateDependenciesInfo() {
+        final StringBuilder output = new StringBuilder()
+
+        for (Dependency dependency : dependencies) {
+            // Only external dependencies are checked
+            if (dependency.group != null && dependency.group.contains("elasticsearch") == false) {
+                output.append(dependency.group).append(':').append(dependency.name).append(',')
+                output.append(dependency.version).append(',')
+                output.append(createURL(dependency.group, dependency.name, dependency.version)).append(',')
+                output.append(getLicenseType(dependency.group, dependency.name)).append('\n')
+            }
+        }
+        outputFile.setText(output.toString(), 'UTF-8')
+    }
+
+    /**
+     * Create an URL on <a href="https://jcenter.bintray.com/">JCenter repository</a>
+     * based on dependency coordinates.
+     */
+    protected String createURL(final String group, final String name, final String version){
+        final String baseURL = 'https://jcenter.bintray.com'
+        return "${baseURL}/${group.replaceAll('\\.' , '/')}/${name}/${version}"
+    }
+
+    /**
+     * Read the LICENSE file associated with the dependency and determine a license type.
+     *
+     * The license type is one of the following values:
+     * <u>
+     *     <li><em>UNKNOWN</em> if LICENSE file is not present for this dependency.</li>
+     *     <li><em>one SPDX identifier</em> if the LICENSE content matches with an SPDX license.</li>
+     *     <li><em>Custom:URL</em> if it' not an SPDX license,
+     *          URL is the Github URL to the LICENSE file in elasticsearch repository.</li>
+     * </ul>
+     *
+     * @param group dependency group
+     * @param name dependency name
+     * @return SPDX identifiers, UNKNOWN or a Custom license
+     */
+    protected String getLicenseType(final String group, final String name) {
+        File license
+
+        if (licensesDir.exists()) {
+            licensesDir.eachFileMatch({ it ==~ /.*-LICENSE.*/ }) { File file ->
+                String prefix = file.name.split('-LICENSE.*')[0]
+                if (group.contains(prefix) || name.contains(prefix)) {
+                    license = file.getAbsoluteFile()
+                }
+            }
+        }
+
+        if (license) {
+            final String content = license.readLines("UTF-8").toString()
+            final String spdx = checkSPDXLicense(content)
+            if (spdx == '') {
```

Comment:
I think it would be cleaner to have the spdx method return null if an spdx license type is not found.

Reply:
Updated.